### ISSUE TITLE
Spring Boot dependency issue with jersey

### ISSUE THREAD

**Issue by Yocker95k** [bonigarcia/selenium-jupiter#86]
# Ignore the first two comments. [Start reading here](https://github.com/bonigarcia/selenium-jupiter/issues/86#issuecomment-703510227)

I want to use the generic driver in a Spring Boot application and set the default browser according to the environment. 
But somehow the setting wont be applied and it always tries to use the chrome-in-docker browser.

I described it on [Stackoverflow ](https://stackoverflow.com/questions/63946640/selenium-jupiter-generic-webdriver-settings-not-applied) 

@bonigarcia Am I doing something wrong?

**Comment by Yocker95k** [comment]
It seems the settings are fine, but something else is wrong.
When I add this dependency 
`endToEndTestImplementation group: 'org.glassfish.hk2', name: 'hk2-api', version: '3.0.0-M2'`
then it works just fine. 

I looked at the DockerService and saw an import of org.glassfish........
And since it was mentioned in my Stacktrace:
```
2020-09-17 23:17:47.196  INFO 14092 --- [    Test worker] d[....path....].SignupPageTest  : Started SignupPageTest in 26.133 seconds (JVM running for 29.179)
    2020-09-17 23:17:47.498  WARN 14092 --- [    Test worker] i.g.bonigarcia.seljup.SeleniumExtension  : Driver handler for context id [engine:junit-jupiter]/[class:de,[....path....].SignupPageTest]/[method:verifyLogInPage(org.openqa.selenium.WebDriver)] not found

    Failed to resolve parameter [org.openqa.selenium.WebDriver genDriver] in method [public void de.[....path....].SignupPageTest.verifyLogInPage(org.openqa.selenium.WebDriver) throws java.lang.Exception]: org/glassfish/hk2/api/MultiException
    org.junit.jupiter.api.extension.ParameterResolutionException: Failed to resolve parameter [org.openqa.selenium.WebDriver genDriver] in method [public void de.[....path....].SignupPageTest.verifyLogInPage(org.openqa.selenium.WebDriver) throws java.lang.Exception]: org/glassfish/hk2/api/MultiException
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:239)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:183)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:144)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:96)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:212)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:208)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:137)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:71)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
        at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
        at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
        at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
        at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
        at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
        at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
        at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
        at com.sun.proxy.$Proxy5.stop(Unknown Source)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:133)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
        at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:832)
    Caused by: java.lang.NoClassDefFoundError: org/glassfish/hk2/api/MultiException
        at io.github.bonigarcia.seljup.DockerService.<init>(DockerService.java:79)
        at io.github.bonigarcia.seljup.SeleniumExtension.initHandlerForDocker(SeleniumExtension.java:381)
        at io.github.bonigarcia.seljup.SeleniumExtension.getDriverHandler(SeleniumExtension.java:242)
        at io.github.bonigarcia.seljup.SeleniumExtension.resolveParameter(SeleniumExtension.java:220)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:216)
        ... 73 more
    Caused by: java.lang.ClassNotFoundException: org.glassfish.hk2.api.MultiException
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 78 more
    
    de.[....path....].SignupPageTest > verifyLogInPage(WebDriver) FAILED
        org.junit.jupiter.api.extension.ParameterResolutionException at ExecutableInvoker.java:239
            Caused by: java.lang.NoClassDefFoundError at DockerService.java:79
                Caused by: java.lang.ClassNotFoundException at BuiltinClassLoader.java:602
```
I added the dependency and suddenly it worked. I dont even have an idea what it is for 🙈 
So now I am not sure if there is something wrong with my dependencies or those transitive dependencies

**Comment by Yocker95k** [comment]
Ok I figured out that applying the Spring Boot plugin causes my problems. 

I made a simple test project 
**build.gradle**
```gradle
plugins {
    id 'org.springframework.boot' version '2.3.4.RELEASE'
    id 'io.spring.dependency-management' version '1.0.10.RELEASE'
    id 'java'
}
dependencies {
    testImplementation('org.springframework.boot:spring-boot-starter-test') {
        exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
    }
    testImplementation("io.github.bonigarcia:selenium-jupiter:3.3.5")
}

```
The plugin removes among others the transitive dependency `+--- org.glassfish.hk2:hk2-api:2.4.0-b34`

**Without the Spring Boot plugin**
```
\--- io.github.bonigarcia:selenium-jupiter:3.3.5
...
...
 +--- com.spotify:docker-client:8.16.0
   ...
   ...
     |    +--- org.glassfish.jersey.core:jersey-client:2.22.2
     |    |    +--- javax.ws.rs:javax.ws.rs-api:2.0.1
     |    |    +--- org.glassfish.jersey.core:jersey-common:2.22.2
     |    |    |    +--- javax.ws.rs:javax.ws.rs-api:2.0.1
     |    |    |    +--- javax.annotation:javax.annotation-api:1.2
     |    |    |    +--- org.glassfish.jersey.bundles.repackaged:jersey-guava:2.22.2
     |    |    |    +--- org.glassfish.hk2:hk2-api:2.4.0-b34
     |    |    |    |    +--- org.glassfish.hk2:hk2-utils:2.4.0-b34
     |    |    |    |    \--- org.glassfish.hk2.external:aopalliance-repackaged:2.4.0-b34
     |    |    |    +--- org.glassfish.hk2.external:javax.inject:2.4.0-b34
     |    |    |    +--- org.glassfish.hk2:hk2-locator:2.4.0-b34
     |    |    |    |    +--- org.glassfish.hk2.external:javax.inject:2.4.0-b34
     |    |    |    |    +--- org.glassfish.hk2.external:aopalliance-repackaged:2.4.0-b34
     |    |    |    |    +--- org.glassfish.hk2:hk2-api:2.4.0-b34 (*)
     |    |    |    |    +--- org.glassfish.hk2:hk2-utils:2.4.0-b34
     |    |    |    |    \--- org.javassist:javassist:3.18.1-GA
     |    |    |    \--- org.glassfish.hk2:osgi-resource-locator:1.0.1
     |    |    +--- org.glassfish.hk2:hk2-api:2.4.0-b34 (*)
     |    |    +--- org.glassfish.hk2.external:javax.inject:2.4.0-b34
     |    |    \--- org.glassfish.hk2:hk2-locator:2.4.0-b34 (*)
     |    +--- org.glassfish.jersey.connectors:jersey-apache-connector:2.22.2
     |    |    +--- org.apache.httpcomponents:httpclient:4.5 -> 4.5.9 (*)
     |    |    +--- org.glassfish.jersey.core:jersey-common:2.22.2 (*)
     |    |    +--- org.glassfish.jersey.core:jersey-client:2.22.2 (*)
     |    |    \--- javax.ws.rs:javax.ws.rs-api:2.0.1
     |    +--- org.glassfish.jersey.media:jersey-media-json-jackson:2.22.2
     |    |    +--- org.glassfish.jersey.core:jersey-common:2.22.2 (*)
     |    |    +--- org.glassfish.jersey.ext:jersey-entity-filtering:2.22.2
     |    |    |    \--- javax.ws.rs:javax.ws.rs-api:2.0.1
     |    |    +--- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.5.4 -> 2.9.8 (*)
     |    |    +--- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.5.4 -> 2.9.8 (*)
     |    |    \--- com.fasterxml.jackson.core:jackson-annotations:2.5.4 -> 2.9.0
```

**With the plugin**
```
\--- io.github.bonigarcia:selenium-jupiter:3.3.5
...
...
   +--- com.spotify:docker-client:8.16.0
   ...
   ...
     |    +--- org.glassfish.jersey.core:jersey-client:2.22.2 -> 2.30.1
     |    |    +--- jakarta.ws.rs:jakarta.ws.rs-api:2.1.6
     |    |    +--- org.glassfish.jersey.core:jersey-common:2.30.1
     |    |    |    +--- jakarta.ws.rs:jakarta.ws.rs-api:2.1.6
     |    |    |    +--- jakarta.annotation:jakarta.annotation-api:1.3.5
     |    |    |    +--- org.glassfish.hk2.external:jakarta.inject:2.6.1
     |    |    |    \--- org.glassfish.hk2:osgi-resource-locator:1.0.3
     |    |    \--- org.glassfish.hk2.external:jakarta.inject:2.6.1
     |    +--- org.glassfish.jersey.connectors:jersey-apache-connector:2.22.2 -> 2.30.1
     |    |    +--- org.apache.httpcomponents:httpclient:4.5.9 -> 4.5.12 (*)
     |    |    +--- org.glassfish.jersey.core:jersey-common:2.30.1 (*)
     |    |    +--- org.glassfish.jersey.core:jersey-client:2.30.1 (*)
     |    |    \--- jakarta.ws.rs:jakarta.ws.rs-api:2.1.6
     |    +--- org.glassfish.jersey.media:jersey-media-json-jackson:2.22.2 -> 2.30.1
     |    |    +--- org.glassfish.jersey.core:jersey-common:2.30.1 (*)
     |    |    +--- org.glassfish.jersey.ext:jersey-entity-filtering:2.30.1
     |    |    |    \--- jakarta.ws.rs:jakarta.ws.rs-api:2.1.6
     |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.10.1 -> 2.11.2
     |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.10.1 -> 2.11.2 (*)
     |    |    \--- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.10.1 -> 2.11.2 (*)
```

**Comment by Yocker95k** [comment]
**I fixed it by** [customizing the managed version of the plugin](https://docs.spring.io/spring-boot/docs/2.1.7.RELEASE/gradle-plugin/reference/html//#managing-dependencies-customizin)

`ext['jersey.version']='2.22.2'`

It is still annoying that this is even necessary. I also noticed that there hasnt been an update for  [spotify/docker-client](https://github.com/spotify/docker-client) in 17 months. Maybe there is an alternative to it?

**Comment by Yocker95k** [comment]
Also having this dependency   
`implementation("org.apache.tika:tika-langdetect:1.24.1")`

leads to this, when running a webdriver with docker:
```java
16:39:34.984 [Test worker] WARN org.apache.cxf.phase.PhaseInterceptorChain - Interceptor for {npipe://localhost:80/containers/create}WebClient has thrown exception, unwinding now
org.apache.cxf.interceptor.Fault: Could not send Message.
	at org.apache.cxf.interceptor.MessageSenderInterceptor.handleMessage(MessageSenderInterceptor.java:49)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
	at org.apache.cxf.jaxrs.client.AbstractClient.doRunInterceptorChain(AbstractClient.java:701)
	at org.apache.cxf.jaxrs.client.WebClient.prepareAsyncClient(WebClient.java:1001)
	at org.apache.cxf.jaxrs.client.WebClient.doInvokeAsync(WebClient.java:964)
	at org.apache.cxf.jaxrs.client.AsyncInvokerImpl.method(AsyncInvokerImpl.java:195)
	at com.spotify.docker.client.DefaultDockerClient.request(DefaultDockerClient.java:2721)
	at com.spotify.docker.client.DefaultDockerClient.createContainer(DefaultDockerClient.java:742)
	at com.spotify.docker.client.DefaultDockerClient.createContainer(DefaultDockerClient.java:724)
	at io.github.bonigarcia.seljup.DockerService.startContainer(DockerService.java:164)
	at io.github.bonigarcia.seljup.handler.DockerDriverHandler.startSelenoidContainer(DockerDriverHandler.java:810)
	at io.github.bonigarcia.seljup.handler.DockerDriverHandler.startDockerBrowser(DockerDriverHandler.java:732)
	at io.github.bonigarcia.seljup.handler.DockerDriverHandler.setHubUrl(DockerDriverHandler.java:334)
	at io.github.bonigarcia.seljup.handler.DockerDriverHandler.getDriverForBrowser(DockerDriverHandler.java:274)
	at io.github.bonigarcia.seljup.handler.DockerDriverHandler.resolve(DockerDriverHandler.java:211)
	at io.github.bonigarcia.seljup.handler.DockerDriverHandler.resolve(DockerDriverHandler.java:189)
	at io.github.bonigarcia.seljup.handler.RemoteDriverHandler.resolve(RemoteDriverHandler.java:104)
	at io.github.bonigarcia.seljup.SeleniumExtension.resolveHandler(SeleniumExtension.java:310)
	at io.github.bonigarcia.seljup.SeleniumExtension.resolveParameter(SeleniumExtension.java:223)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:216)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:183)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:144)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:96)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:212)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:208)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:137)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:71)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:133)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.net.MalformedURLException: unknown protocol: npipe
	at java.base/java.net.URL.<init>(URL.java:679)
	at java.base/java.net.URL.fromURI(URL.java:746)
	at java.base/java.net.URI.toURL(URI.java:1139)
	at org.apache.cxf.transport.http.Address.getURL(Address.java:60)
	at org.apache.cxf.transport.http.URLConnectionHTTPConduit.createConnection(URLConnectionHTTPConduit.java:112)
	at org.apache.cxf.transport.http.URLConnectionHTTPConduit.setupConnection(URLConnectionHTTPConduit.java:125)
	at org.apache.cxf.transport.http.HTTPConduit.prepare(HTTPConduit.java:505)
	at org.apache.cxf.interceptor.MessageSenderInterceptor.handleMessage(MessageSenderInterceptor.java:47)
	... 92 common frames omitted
16:39:34.990 [Test worker] DEBUG org.apache.cxf.phase.PhaseInterceptorChain - Adding interceptor org.apache.cxf.jaxrs.client.AbstractClient$ConnectionFaultInterceptor@605050cf to phase pre-stream
16:39:34.990 [Test worker] DEBUG org.apache.cxf.phase.PhaseInterceptorChain - Chain org.apache.cxf.phase.PhaseInterceptorChain@36f799d0 was created. Current flow:
  pre-stream [ConnectionFaultInterceptor]
```

**Comment by bonigarcia** [comment]
It seems you solve your initial problem, isn't it?

Regarding `unknown protocol: npipe, no clue`, sorry. Did you solve it?

**Comment by Yocker95k** [comment]
Sort of. I had to exclude the dependency `implementation("org.apache.tika:tika-langdetect:1.24.1")`  from the configuration for Selenium-Jupiter tests. 

I think its the docker clients fault. At least my first issue was its fault. It is outdated and an inactive repository. Maybe [this fork ](https://github.com/dmandalidis/docker-client) is an alternative.

**Comment by bonigarcia** [comment]
Yes, I want to change Spotify docker-client to another library, maybe [docker-java](https://github.com/docker-java/docker-java) (or maybe the fork you pointed out, thanks for that).

**Comment by bonigarcia** [comment]
This should be fixed in version 3.4.0, just released.
